src/webgpu/api/operation/buffers/map_detach.spec.ts

```diff
@@ -19,53 +23,51 @@ class F extends GPUTest {
+    - Tests {mappable, unmappable mapAtCreation, mappable mapAtCreation}
+    - Tests while {mapped, mapped at creation, mapped at creation then unmapped and mapped again}`
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combine('mappedAtCreation', [false, true])
+      .combineWithParams([
+        { usage: GPUConst.BufferUsage.COPY_SRC },
+        { usage: GPUConst.BufferUsage.MAP_WRITE | GPUConst.BufferUsage.COPY_SRC },
+        { usage: GPUConst.BufferUsage.COPY_DST | GPUConst.BufferUsage.MAP_READ },
+        {
+          usage: GPUConst.BufferUsage.MAP_WRITE | GPUConst.BufferUsage.COPY_SRC,
+          mapMode: GPUConst.MapMode.WRITE,
+        },
+        {
+          usage: GPUConst.BufferUsage.COPY_DST | GPUConst.BufferUsage.MAP_READ,
+          mapMode: GPUConst.MapMode.READ,
+        },
+      ])
+      .combineWithParams([
+        { unmap: true, destroy: false },
+        { unmap: false, destroy: true },
+        { unmap: true, destroy: true },
+      ])
+      .unless(p => p.mappedAtCreation === false && p.mapMode === undefined)
+  )
   .fn(async t => {
-    const buffer = t.device.createBuffer({ size: 4, usage: GPUBufferUsage.MAP_WRITE });
-    await buffer.mapAsync(GPUMapMode.WRITE);
-    const arrayBuffer = buffer.getMappedRange();
-    t.checkDetach(buffer, arrayBuffer, t.params.unmap, t.params.destroy);
-  });
-
-g.test('mapAsync,read')
-  .paramsSimple([
-    { unmap: true, destroy: false }, //
-    { unmap: false, destroy: true },
-    { unmap: true, destroy: true },
-  ])
-  .fn(async t => {
-    const buffer = t.device.createBuffer({ size: 4, usage: GPUBufferUsage.MAP_READ });
-    await buffer.mapAsync(GPUMapMode.READ);
-    const arrayBuffer = buffer.getMappedRange();
-    t.checkDetach(buffer, arrayBuffer, t.params.unmap, t.params.destroy);
-  });
-
-g.test('create_mapped')
-  .paramsSimple([
-    { unmap: true, destroy: false },
-    { unmap: false, destroy: true },
-    { unmap: true, destroy: true },
-  ])
-  .fn(async t => {
-    const desc = {
-      mappedAtCreation: true,
+    const { usage, mapMode, mappedAtCreation, unmap, destroy } = t.params;
+    const buffer = t.device.createBuffer({
       size: 4,
-      usage: GPUBufferUsage.MAP_WRITE,
-    };
-    const buffer = t.device.createBuffer(desc);
-    const arrayBuffer = buffer.getMappedRange();
+      usage,
+      mappedAtCreation,
+    });
 
-    const view = new Uint8Array(arrayBuffer);
-    t.expect(arrayBuffer.byteLength === 4);
-    t.expect(view.length === 4);
+    if (mapMode !== undefined) {
+      if (mappedAtCreation) {
+        buffer.unmap();
+      }
+      await buffer.mapAsync(mapMode);
+    }
 
-    if (t.params.unmap) buffer.unmap();
-    if (t.params.destroy) buffer.destroy();
-    t.expect(arrayBuffer.byteLength === 0, 'ArrayBuffer should be detached');
-    t.expect(view.byteLength === 0, 'ArrayBufferView should be detached');
+    const arrayBuffer = buffer.getMappedRange();
+    t.checkDetach(buffer, arrayBuffer, unmap, destroy);
```

Comment:
totally optional nit: If this is the only usage of this function now, it could be inlined here.